frontend/src/modules/colorLabel/views/colorLabelPopupView.js

```diff
@@ -0,0 +1,97 @@
+define([
+    'core/origin',
+    'core/views/originView'
+], function(Origin, OriginView) {
+
+    var ColorLabelPopUpView = OriginView.extend({
+
+        className: 'colorlabel',
+
+        events: {
+            'click .reset': 'onReset',
+            'click .apply': 'onApply',
+            'click .cancel': 'remove',
+            'click .color-item': 'onItemClick'
+        },
+
+        initialize: function(options) {
+            this.parentView = options.parentView;
+            OriginView.prototype.initialize.apply(this, arguments);
+        },
+
+        onItemClick: function(event) {
+            var $elm = $(event.currentTarget);
+            this.selected = $elm.data('colorlabel');
+            this.updateClasses();
+        },
+
+        updateClasses: function() {
+            this.$('.color-item').removeClass('selected');
+            this.$('[data-colorlabel="'+this.selected+'"]').addClass('selected');
+        },
+
+        onReset: function() {
+            var model = this.parentView.model;
+            this.selected = null;
+
+            model.save('_colorLabel', '', {
+                patch: true,
+                success: _.bind(this.onSuccess, this),
+                error: _.bind(this.onError, this)
+            });
+        },
+
+        onApply: function(event) {
+            if (!this.selected) return;
+            this.addItem();
+        },
+
+        preRender: function() {
+            this.selected = null;
```

Comment:
`this.selected` can be declared on instantiation if you preferred to not use this function for it, like [`className`](https://github.com/adaptlearning/adapt_authoring/pull/1921/files#diff-4082fa8f4c93488e4a000c7ea57ed2dfR8) is:
```js
selected: null
```